storage/namespace.go

```diff
@@ -568,9 +568,7 @@ func (n *dbNamespace) Flush(
 func (n *dbNamespace) NeedsFlush(startInclusive time.Time, endInclusive time.Time) bool {
 	var (
 		blockSize   = n.nopts.RetentionOptions().BlockSize()
-		startBlock  = startInclusive.Truncate(blockSize)
-		endBlock    = endInclusive.Truncate(blockSize)
-		blockStarts = timesInRange(startBlock, endBlock, blockSize)
+		blockStarts = timesInRange(startInclusive, endInclusive, blockSize)
```

Comment:
Ok, that is pretty intense.  I hope I remember that next time I work with these APIs... lol.